src/Core/Domain/CartRule/ValueObject/DiscountApplicationType.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\CartRule\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\CartRule\Exception\CartRuleConstraintException;
+
+/**
+ * Discount application type
+ */
+class DiscountApplicationType
```

Comment:
I'm not sure as well, I followed the naming in the interface, where it's called cart rule "Actions". @matks wdyt?

Reply:
Mmmmm ... this is a hard question.
I think @sarjon suggestion is right from a semantic point of view, but @rokaszygmantas did the right choice by sticking to legacy name as it will ease people to understand what it means.

I think we should stick to Actions for this reason 👍Action is "good enough" and helps people acquiring this new vocabulary we are building in Domain folder